src/_pytest/fixtures.py

```diff
@@ -786,13 +786,21 @@ def fail_fixturefunc(fixturefunc, msg):
 def call_fixture_func(fixturefunc, request, kwargs):
     yieldctx = is_generator(fixturefunc)
     if yieldctx:
-        it = fixturefunc(**kwargs)
-        res = next(it)
-        finalizer = functools.partial(_teardown_yield_fixture, fixturefunc, it)
-        request.addfinalizer(finalizer)
+        generator = fixturefunc(**kwargs)
+        try:
+            fixture_return_value = next(generator)
```

Comment:
"fixture return value" might be a bit confusing in the generator case, because a generator does actually have a return value, which is different from the yielded value(s). My suggestion is `fixture_result`.